Signal/src/call/PeerConnectionClient.swift

```diff
@@ -1092,6 +1110,90 @@ class HardenedRTCSessionDescription {
     }
 }
 
+protocol VideoCaptureSettingsDelegate: class {
+    var videoWidth: Int32 { get }
+    var videoHeight: Int32 { get }
+}
+
+class VideoCaptureController {
+
+    let capturer: RTCCameraVideoCapturer
+    weak var settingsDelegate: VideoCaptureSettingsDelegate?
+    var isUsingFrontCamera: Bool = true
+
+    public init(capturer: RTCCameraVideoCapturer, settingsDelegate: VideoCaptureSettingsDelegate) {
+        self.capturer = capturer
+        self.settingsDelegate = settingsDelegate
+    }
+
+    public func startCapture() {
+        let position: AVCaptureDevice.Position = isUsingFrontCamera ? .front : .back
+        guard let device: AVCaptureDevice = self.device(position: position) else {
+            owsFail("unable to find captureDevice")
+            return
+        }
+
+        guard let format: AVCaptureDevice.Format = self.format(device: device) else {
+            owsFail("unable to find captureDevice")
+            return
+        }
+
+        let fps = self.framesPerSecond(format: format)
+
+        capturer.startCapture(with: device, format: format, fps: fps)
+    }
+
+    public func stopCapture() {
+        self.capturer.stopCapture()
```

Comment:
why all of this unnecessary `self.`?  NBD.

Reply:
I'm personally not a fan of the "implicit self", as I find it makes it harder to distinguish from a local variable. 

I would prefer that we always add the explicit self, even when not required. But I don't feel that strongly about it and am happy to conform if you have a preference for the implicit self.